Refresh.Interfaces.APIv3/Endpoints/DataTypes/Response/Categories/ApiUserCategoryResponse.cs

```diff
@@ -0,0 +1,51 @@
+using Bunkum.Core;
+using Refresh.Core.Types.Categories.Users;
+using Refresh.Core.Types.Data;
+using Refresh.Database;
+using Refresh.Database.Models.Users;
+using Refresh.Interfaces.APIv3.Endpoints.DataTypes.Response.Users;
+
+namespace Refresh.Interfaces.APIv3.Endpoints.DataTypes.Response.Categories;
+
+[JsonObject(NamingStrategyType = typeof(CamelCaseNamingStrategy))]
+public class ApiUserCategoryResponse : ApiCategoryResponse, IApiResponse, IDataConvertableFrom<ApiUserCategoryResponse, GameUserCategory>
+{
+    public required ApiGameUserResponse? PreviewItem { get; set; }
+    
+    public static ApiUserCategoryResponse? FromOld(GameUserCategory? old, GameUser? PreviewItem,
+        DataContext dataContext)
+    {
+        if (old == null) return null;
+
+        return new ApiUserCategoryResponse
+        {
+            Name = old.Name,
+            Description = old.Description,
+            IconHash = old.IconHash,
+            FontAwesomeIcon = old.FontAwesomeIcon,
+            ApiRoute = old.ApiRoute,
+            RequiresUser = old.RequiresUser,
+            PreviewItem = ApiGameUserResponse.FromOld(PreviewItem, dataContext),
+            Hidden = old.Hidden,
+        };
```

Comment:
To reduce duplication, could `ApiCategoryResponse` be `ApiCategoryResponse<TPreviewItem>` and then just have a generic function for mapping the item? Fine as-is but I like to think about other paths we could take.